File: PS1/nhe3/Diagonal.py
'''
************************************************
  ____ ____    _  ___   __
 / ___/ ___|  / |/ _ \ / /_
| |   \___ \  | | (_) | '_ \
| |___ ___) | | |\__, | (_) |
 \____|____/  |_|  /_/ \___/


Problem set 1


Question 2

Given a square matrix size N x N, calculate the product of the sums across the two main diagonals.
For example, given the input:

3

4   5   7
3   1   5
9   3   2

Return: (4+1+2)*(9+1+7) = 119

PLEASE LOOK AT PS1.txt FOR MORE DETAILS!!!

************************************************
'''
print (int('4'))
def diagonal(input):
    # YOUR CODE HERE
    diagList1 = []
    diagList2 = []
    for i in range(len(input)):
        diagList1.append(int(input[i][i]))
        diagList2.append(int(input[i][len(input)-i-1]))
    print (diagList1)
    print (diagList2)
    return sum(diagList1) * sum(diagList2)

try:
    with open('Diagonal.txt') as f:
        data = f.read().split()
        num_of_tests = data[0]
        data = data[1:]

        for i in range(0, int(num_of_tests)):
            matrix = []
            dimen = int(data[0])
            data = data[1:]

            for j in range(0, dimen):
                row = data[:dimen]
                matrix.append(row)
                data = data[dimen:]
            print (diagonal(matrix))
except IOError(e):
    print (e)

diagonal([[1,2],[3,4]])

File: PS1/rseligm2/Diagonal.java
import java.util.*;
import java.io.*;

/*
************************************************
  ____ ____    _  ___   __   
 / ___/ ___|  / |/ _ \ / /_  
| |   \___ \  | | (_) | '_ \ 
| |___ ___) | | |\__, | (_) |
 \____|____/  |_|  /_/ \___/ 
                             

Problem set 1


Question 2

Given a square matrix size N x N, calculate the product of the sums across the two main diagonals.
For example, given the input:

3

4	5	7
3	1	5
9	3	2

Return: (4+1+2)*(9+1+7) = 119

PLEASE LOOK AT PS1.txt FOR MORE DETAILS!!!

************************************************

*/
public class Diagonal {

	public static int diagonal(int[][] input) {
		int product;
		int leftdown=0;
		int rightdown=0;
		int matrixSize=input.length;
		int[] leftdownarray=new int[matrixSize];
		int[] rightdownarray=new int[matrixSize];
		int y=0;
		for(int x=0;x<matrixSize;x++)
		{
			leftdownarray[x]=input[x][y];
			leftdown+=input[x][y];
			y++;
		}
		y=0;
		for(int xx=matrixSize-1;xx>=0;xx--)
		{
			rightdownarray[y]=input[xx][y];
			rightdown+=input[xx][y];
			y++;
		}
		product=rightdown*leftdown;
		
		
		//YOUR CODE HERE
		return product;
	}





	public static void main(String[] args) {
		File file = new File("Diagonal.txt");
		try {
			Scanner scan = new Scanner(file);
			int numberOfCases = scan.nextInt();
			for(int i = 0; i < numberOfCases; i++) {
				int N = scan.nextInt();
				int[][] matrix = new int[N][N];
				for(int j = 0; j < N; j++) {
					for(int k = 0; k < N; k++) {
						matrix[j][k] = scan.nextInt();
					}
				}
				System.out.println(diagonal(matrix));
			}
			scan.close();
		}
		catch (FileNotFoundException e) {
			e.printStackTrace();
		}
	}

}